lib/gpio/pin.toit

```diff
@@ -91,11 +101,60 @@ class Pin:
   If $open_drain is true, the output configuration will use
     - pull-low for 0
     - open-drain for 1
+
+  Deprecated. Use $configure instead. Note that $configure behaves differently than
+    $config when a pin was initialized with a pull-up or pull-down resistor. This function
+    ($config) maintains the pull-up/pull-down configuration of the pin. However, $configure
+    resets that configuration.
   */
+  // When removing this function, it's safe to remove `pull_down_` and `pull_up_` as well.
   config --input/bool=false --output/bool=false --open_drain/bool=false:
     if open_drain and not output: throw "INVALID_ARGUMENT"
     gpio_config_ num (input and pull_up_) (input and pull_down_) input output open_drain
 
+  /**
+  Changes the configuration of this pin.
+
+  If $input is true, the pin is configured as an input.
+  If $output is true, the pin is configured as an output. If $open_drain is set, then the pin
+    value is set to 1 (not pulling to ground). Otherwise the pin outputs 0.
+
+  It is safe to use a pin as $input and $output at the same time, but typically this
+    requires the $open_drain flag.
+
+  If a pin is used as $input and $output without $open_drain, then the
+    pin can only read the value that was set with $set. It can/should not read a
+    value that was set by the outside. In fact, doing so could damage the microcontroller, as
+    the external device would need to short circuit the pin.
+
+  If a pin is configured to be an input, it can have a $pull_up or $pull_down.
+
+  If $open_drain is set, then the pin can only pull the pin to the ground. Together, with
+    a $pull_up resistor this still allows the pin to emit both 0 and 1. In this configuration,
+    connected devices can also safely pull the pin to ground without damaging the microcontroller.
+    This configuration is typically used in communications that only use one data bus for
+    input and output, such as the DHT11/DHT22, the i2c bus, and the one-wire bus. Note, that
+    the corresponding libraries (like the i2c library) already take care of setting this
+    configuration for you.
+  Note that it is not safe to ground an $open_drain pin and to connect it externally to VCC.
```

Comment:
When do you add a newline between the paragraphs and when do you not?

Reply:
I don't think there is a big rule, and the toitdoc currently doesn't make a difference.
Here, the "Note" is related to the 'open_drain' so I kept it together, but you are right: the next note should be attached too.
done.